src/Adapter/Alias/Repository/AliasRepository.php

```diff
@@ -199,4 +199,26 @@ public function deleteAliasesBySearchTerm(string $searchTerm): void
             );
         }
     }
+
+    /**
+     * @param string $searchPhrase
+     * @param int|null $limit
+     *
+     * @return array
+     */
+    public function searchAliases(string $searchPhrase, ?int $limit = null): array
+    {
+        $dbSearchPhrase = sprintf('"%%%s%%"', $searchPhrase);
+        $qb = $this->connection->createQueryBuilder();
+
+        return $qb
+            ->addSelect('a.search')
+            ->from($this->dbPrefix . 'alias', 'a')
+            ->addOrderBy('a.search', 'ASC')
+            ->addGroupBy('a.search')
+            ->setMaxResults($limit)
+            ->where($qb->expr()->like('a.search', $dbSearchPhrase))
```

Comment:
I think you can set parameters like

```suggestion
            ->where($qb->expr()->like('a.search', ':searchPhrase'))
            ->setParameter('searchPhrase', '%' . $searchPhrase. '%')
```
Im not  even sure if LIKE expression is that much usefull, you could as well use simply
    ->where('a.search LIKE :searchPhrase')
    ->setParameter('searchPhrase', '%' . $searchPhrase. '%')
    
    in this case you can get rid of the $qb variable as it would not be needed anymore.
  
  Though both cases works so I don't mind... choose whatver you think is better :smile: